test/specs/rules/disallow-quoted-keys-in-objects.js

```diff
@@ -67,4 +67,12 @@ describe('rules/disallow-quoted-keys-in-objects', function() {
 
         assert(checker.checkString('var x = { "null": 1, undefined: "foo" }').isEmpty());
     });
+
+    it('should report non-reserved quoted keys when "allButReserved" mode is used. #1669', function() {
```

Comment:
@markelog Something like that: https://github.com/zxqfox/node-jscs/commit/fbd1e86a6d00e0dc44dc58fcf161bba924d80613